From 9c20edaf3dc726a8c94e8edfed2ce7fed4bb98f4 Mon Sep 17 00:00:00 2001
From: "Cao,Bing Bu" <bingbu.cao@intel.com>
Date: Wed, 19 Nov 2014 09:54:07 +0000
Subject: [PATCH 09/10] Correct the ad5823 VCM focus config

In current ad5823 driver, there is a mistake in
vcm focus config.
As the camera HAL has been updated,driver need
no more do (MAX-val) for focus position.
This patch corrects this mistake.

Change-Id: I1a3e6403e78c09b429ee2fa49448066756685ac6
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-3862
Signed-off-by: Cao,Bing Bu <bingbu.cao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/29251
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Jing Tao <jing.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
Reviewed-by: Andrew J Ross <andrew.j.ross@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/i2c/ov5693-ecs/ov5693.c   |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
index 40272d0..ed13182 100755
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -925,7 +925,7 @@ int ad5823_t_focus_abs(struct v4l2_subdev *sd, s32 value)
 	int ret;
 
 	value = min(value, AD5823_MAX_FOCUS_POS);
-	ret = ad5823_t_focus_vcm(sd, AD5823_MAX_FOCUS_POS - value);
+	ret = ad5823_t_focus_vcm(sd, value);
 
 	return ret;
 }
@@ -1277,8 +1277,7 @@ static int ov5693_init(struct v4l2_subdev *sd)
 	/*change initial focus value for ad5823*/
 	if(dev->vcm == VCM_AD5823) {
 		dev->focus = AD5823_INIT_FOCUS_POS;
-		ov5693_t_focus_abs(sd, (AD5823_MAX_FOCUS_POS
-					- AD5823_INIT_FOCUS_POS));
+		ov5693_t_focus_abs(sd, AD5823_INIT_FOCUS_POS);
 	} else {
 		dev->focus = 0;
 		ov5693_t_focus_abs(sd, 0);
-- 
1.7.9.5

